src/service/PreconditionService/PreconditionsService.Types.ts

```diff
@@ -0,0 +1,3 @@
+export interface Precondition {
+    satisfied: boolean;
+}
```

Comment:
Working with a namespace will be better for services, because somethimes we have overlaps with the naming

```suggestion
export namespace PreconditionsServiceTypes {
   export interface Precondition {
       satisfied: boolean;
   }
}
```